Easi 3248/request edits email

StevenWadeOddball opened the pull request:
# EASI-3248

## Changes and Description

- Create Request Edits Email
    - Create System Intake Edits Requested Email
       - New struct on email client to separate calls to IT gov v2 emails


## How to test this change
Use postman to do the following ( or use the front end)
1. Create a system intake and submit it
2. Request edits to the system intake
3. Confirm that you receive an email in mail catcher
4. Send another edits requested email that does not include additionalInfo. 
5. Confirm that the additional information section is left out of the email


Note, using the listAllRecipients method does introduce the ability to send to the IT Governance and IT Investment inbox. If desired we could remove the ability to send to the It investment inbox, but it would be ignoring the properties provided on the `EmailNotificationRecipients` type.
## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Tested user-facing changes with voice-over and the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)


## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

ClayBenson94 commented:
@StevenWadeOddball I pulled in the CRA -> Vite changes into this PR to ensure that folks don't have to do as much `yarn install` when switching branches. No conflicts, so things should be all set, but wanted to give a heads up!

ClayBenson94 reviewed (changes requested):
@StevenWadeOddball Would you mind adding a manual test to https://github.com/CMSgov/easi-app/blob/1bac362ddb1eb451aa8bffd455c6b0fb2d53f996/cmd/test_email_templates/main.go ? 


Otherwise, the changes seem good to me!
![image](https://github.com/CMSgov/easi-app/assets/15203744/7c120c34-3ebe-4523-ad9d-767b802c0e40)


While not necessary for functionality, it's helpful to keep this file up-to-date with our emails so we can quickly validate changes to them (styling, templating, etc.)

StevenWadeOddball commented on pkg/email/request_edits_test.go:
```diff
@@ -0,0 +1,61 @@
+	}
+	client, err := NewClient(s.config, &sender)
+	s.NoError(err)
+	// expectedEmail :=
```
Thanks Clay, I removed it.

StevenWadeOddball commented:
Thanks @ClayBenson94 for telling me about the template program! I updated and added an entry for the new email. I also added a command `scripts/dev email` to call the program.
